lib/rules/no-unused-vars.js

```diff
@@ -155,7 +162,14 @@ module.exports = {
          * @returns {UnusedVarMessageData} The message data to be used with this unused variable.
          */
         function getAssignedMessageData(unusedVar) {
-            const additional = config.varsIgnorePattern ? `. Allowed unused vars must match ${config.varsIgnorePattern.toString()}` : "";
+            const def = unusedVar.defs && unusedVar.defs[0];
+            let additional = "";
+
+            if (config.destructuredArrayIgnorePattern && def.name.parent.type === "ArrayPattern") {
```

Comment:
```suggestion
            if (config.destructuredArrayIgnorePattern && def && def.name.parent.type === "ArrayPattern") {
```

To make this function safe in case a variable with zero definitions is passed to it.